rotkehlchen/chain/evm/decoding/zerox/decoder.py

```diff
@@ -65,6 +67,43 @@ def _update_send_and_receive_events(
             receive_event.event_type = HistoryEventType.TRADE
             receive_event.event_subtype = HistoryEventSubType.RECEIVE
             receive_event.notes = f'Receive {receive_event.balance.amount} {receive_event.asset.symbol_or_name()} as the result of a swap via the 0x protocol'  # noqa: E501
+        if fee_event is not None:
+            fee_event.counterparty = CPT_ZEROX
+            fee_event.address = self.router_address
+            fee_event.event_type = HistoryEventType.SPEND
+            fee_event.event_subtype = HistoryEventSubType.FEE
+            fee_event.notes = f'Spend {fee_event.balance.amount} {fee_event.asset.symbol_or_name()} as a 0x protocol fee'  # noqa: E501
+
+    def _merge_similar_events(
+            self,
+            decoded_events: 'list[EvmEvent]',
+            send_events: 'list[EvmEvent]',
+            receive_events: 'list[EvmEvent]',
+            fee_event: 'EvmEvent | None' = None,
+    ) -> None:
+        """Sum the balances of the events, updates them with the 0x values, replace them with the
+        Merged Events in decoded_events list, and maybe shuffle them with proper order."""
```

Comment:
Can you in the docstring point that this is done because 0x can split the trades in multiple routes?